client/verta/verta/_dataset_versioning/dataset_version.py

```diff
@@ -413,3 +442,146 @@ def base_path(self):
             " considering accessing the paths of specific components"
             " using `list_components()[i].path` instead",
         )
+
+    # The following properties are for managed
+    # TODO: consolidate this with similar method in `_ModelDBEntity`
+    def _get_url_for_artifact(self, dataset_component_path, method, part_num=0):
```

Comment:
The rest of this file (`_get_url_for_artifact` and `_upload_artifact`) are copied directly from `Commit`; the only changes are for the URLs and the protos.